modules/jpa-spring-data/src/main/java/io/oasp/module/jpa/dataaccess/impl/data/GenericRepositoryImpl.java

```diff
@@ -0,0 +1,94 @@
+package io.oasp.module.jpa.dataaccess.impl.data;
+
+import java.io.Serializable;
+
+import javax.persistence.EntityManager;
+import javax.persistence.LockModeType;
+
+import org.springframework.data.jpa.repository.support.JpaEntityInformation;
+import org.springframework.data.jpa.repository.support.SimpleJpaRepository;
+
+import com.querydsl.core.alias.Alias;
+import com.querydsl.core.types.EntityPath;
+import com.querydsl.jpa.impl.JPADeleteClause;
+import com.querydsl.jpa.impl.JPAQuery;
+
+import io.oasp.module.jpa.dataaccess.api.data.GenericRepository;
+
+/**
+ * Implementation of {@link GenericRepository} based on {@link SimpleJpaRepository}. All repository interfaces derived
+ * from {@link GenericRepository} will be based on this implementation at runtime.
+ *
+ * @param <E> generic type of the managed {@link #getEntityClass() entity}.
+ * @param <ID> generic type of the {@link net.sf.mmm.util.entity.api.PersistenceEntity#getId() primary key} of the
+ *        entity.
+ *
+ * @since 3.0.0
+ */
+// QueryDslJpaRepository<E, ID> forces you to use QueryDSL APT generation what is not desired
+// therefore no support for QueryDslPredicateExecutor (we offer more flexible QueryDSL support anyhow)
```

Comment:
We should put this remark into javadocs. Otherwise, we might get features requests for this as the docs were not accessible easily.

Reply:
fine. I will do that.